src/languages/crystal.js

```diff
@@ -4,11 +4,12 @@ Author: TSUYUSATO Kitsune <make.just.on@gmail.com>
 */
 
 function(hljs) {
-  var NUM_SUFFIX = '(_[uif](8|16|32|64))?';
+  var INT_SUFFIX = '(_*[ui](8|16|32|64))?';
+  var FLOAT_SUFFIX = '(_*f(32|64))?';
   var CRYSTAL_IDENT_RE = '[a-zA-Z_]\\w*[!?=]?';
-  var RE_STARTER = '!=|!==|%(?!})|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|/|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|' +
-    '>>|>|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~';
-  var CRYSTAL_METHOD_RE = '[a-zA-Z_]\\w*[!?=]?|[-+~]\\@|<<|>>|=~|===?|<=>|[<>]=?|\\*\\*|[-/+%^&*~`|]|\\[\\][=?]?';
+  var RE_STARTER = '!=|!==|%(?!})|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|-|-=|/=|:|;|<<|<<=|<=|<|===|==|=|>>>=|>>=|>=|>>>|' +
```

Comment:
Removing this and using `hljs.RE_STARTERS_RE` are preferred. Ruby grammar uses it.

Reply:
Q&A:

> Why `hljs.RE_STARTERS_RE` is preferred?

It is standard way and good enough.

> However, it reproduce missing macro ending problem like previous PR.

It can prevent by following:

```javascript
    begin: '(?!%})(' + hljs.RE_STARTERS_RE + ')\\s*',
```

- - -

I believe it is no problem using `hljs.RE_STARTERS_RE` instead.

Reply:
`hljs.RE_STARTERS_RE` includes `%` which messes up macro expansions. We need to have `%(?!})` instead. See https://github.com/highlightjs/highlight.js/pull/1875#issuecomment-431299162

Reply:
Doesn't the solution proposed by @MakeNowJust work?

> It can prevent by following:
>
>        begin: '(?!%})(' + hljs.RE_STARTERS_RE + ')\\s*',